### PR TITLE
RUMM-2033 tvOS Compatibility

### PR DESCRIPTION
**Pull request by maxep**
### What and why?

Add compilation and runtime flags for tvOS.

### How?

This PR only focuses on making codebase compatible with tvOS. Targets definition will come next.

### Review checklist

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing change.

### CODE REVIEW

File: Sources/Datadog/Core/System/CarrierInfoProvider.swift [hunk 1]
```diff
@@ -48,7 +50,7 @@ internal protocol CarrierInfoProviderType {
 extension CarrierInfo.RadioAccessTechnology {
     init(ctRadioAccessTechnologyConstant: String) {
         switch ctRadioAccessTechnologyConstant {
-        #if !targetEnvironment(macCatalyst)
+        #if os(iOS) && !targetEnvironment(macCatalyst)
```
Comment (fuzzybinary) [inline Sources/Datadog/Core/System/CarrierInfoProvider.swift:53]: Do we need that `!targetEnvironment(macCatalyst)` if we're checking specifically for iOS?

**Review by fuzzybinary — APPROVED** [review]

File: Sources/Datadog/Core/System/CarrierInfoProvider.swift [hunk 2]
```diff
@@ -98,16 +100,13 @@ internal class CarrierInfoProvider: CarrierInfoProviderType {
     }
 }
 
-#if targetEnvironment(macCatalyst)
-
-/// Dummy provider for Mac Catalyst which doesn't support carrier info.
-internal struct MacCatalystCarrierInfoProvider: CarrierInfoProviderType {
+/// Dummy provider for platforms which doesn't support carrier info.
+internal struct NOPCarrierInfoProvider: CarrierInfoProviderType {
```
Comment (ncreated) [inline Sources/Datadog/Core/System/CarrierInfoProvider.swift:104]: 👍 for this renaming

File: Sources/Datadog/FeaturesIntegration/WebView/WKUserContentController+Datadog.swift [hunk 3]
```diff
@@ -4,7 +4,7 @@
  * Copyright 2019-2020 Datadog, Inc.
  */
 
-#if DD_SDK_ENABLE_EXPERIMENTAL_APIS
+#if !os(tvOS) && DD_SDK_ENABLE_EXPERIMENTAL_APIS
```
Comment (ncreated) [inline Sources/Datadog/FeaturesIntegration/WebView/WKUserContentController+Datadog.swift:7]: This basically means "no WebView tracking" / "sub apps" for tvOS. Do I see it right in the docs that [`WebKit` is unavailable on tvOS](https://developer.apple.com/documentation/webkit)?

**Review by ncreated — APPROVED** [review]
Looks good 👌.

The number of compiler conditions and platform/env branches now becomes worrying. To ensure that everything compiles and works fine on all platforms (iOS + tvOS), it would be great to consider adding smoke + unit tests for tvOS in `RUMM-1651`'s scope.

File: Sources/Datadog/Core/System/CarrierInfoProvider.swift (hunk 1 above)
Comment (maxep) [reply Sources/Datadog/Core/System/CarrierInfoProvider.swift:53]: Indeed, good catch 👍

File: Sources/Datadog/FeaturesIntegration/WebView/WKUserContentController+Datadog.swift (hunk 3 above)
Comment (maxep) [reply Sources/Datadog/FeaturesIntegration/WebView/WKUserContentController+Datadog.swift:7]: Yes, WebKit is not supported on tvOS so hybrid apps are not possible on this platform.

**Comment by maxep** [comment]
@ncreated Yes, unit tests are coming next in #766.
This PR is a chunk of #763 which successfully builds and tests (units) for tvOS. Tho I need to add tasks for integration and dependency managers tests!

**Pull request merged**